pkg/apis/core/types_shoot.go

```diff
@@ -147,8 +147,46 @@ type ShootStatus struct {
 	AdvertisedAddresses []ShootAdvertisedAddress
 	// MigrationStartTime is the time when a migration to a different seed was initiated.
 	MigrationStartTime *metav1.Time
+	// Credentials contains information about the shoot credentials.
+	Credentials *ShootCredentials
 }
 
+// ShootCredentials contains information about the shoot credentials.
+type ShootCredentials struct {
+	// Rotation contains information about the credential rotations.
+	Rotation *ShootCredentialsRotation
+}
+
+// ShootCredentialsRotation contains information about the certificate authority credential rotation.
+type ShootCredentialsRotation struct {
+	// CertificateAuthorities contains information about the certificate authority credential rotation.
+	CertificateAuthorities *ShootCARotation
+}
+
+// ShootCARotation contains information about the certificate authority credential rotation.
+type ShootCARotation struct {
+	// Phase describes the phase of the certificate authority credential rotation.
+	Phase ShootCredentialsRotationPhase
+	// LastCompletionTime is the most recent time when the certificate authority credential rotation was successfully
+	// completed.
+	LastCompletionTime *metav1.Time
+}
+
+// ShootCredentialsRotationPhase is a string alias.
+type ShootCredentialsRotationPhase string
+
+const (
+	// RotationPrepare is a constant for the credentials rotation phase describing that the procedure is being prepared.
+	RotationPrepare ShootCredentialsRotationPhase = "Prepare"
+	// RotationPrepared is a constant for the credentials rotation phase describing that the procedure was prepared.
+	RotationPrepared ShootCredentialsRotationPhase = "Prepared"
+	// RotationComplete is a constant for the credentials rotation phase describing that the procedure is being
+	// completed.
+	RotationComplete ShootCredentialsRotationPhase = "Complete"
+	// RotationCompleted is a constant for the credentials rotation phase describing that the procedure was completed.
+	RotationCompleted ShootCredentialsRotationPhase = "Completed"
```

Comment:
I think it is easy to confuse `prepare` with `prepared` and `complete` with `completed`. Also `complete` sounds as the phase is already finished.
```suggestion
	//PhaseOneInProgress is a constant for the credentials rotation phase describing that Gardener is currently in the process of executing phase one.
	PhaseOneInProgress ShootCredentialsRotationPhase = "PhaseOneInProgress"
	// PhaseOneDone is a constant for the credentials rotation phase describing that Gardener finished all actions required from Gardener side in step one.
	PhaseOneDone ShootCredentialsRotationPhase = "PhaseOneDone"
	// PhaseTwoInProgress is a constant for the credentials rotation phase describing that Gardener is currently in the process of executing phase two.
	PhaseTwoInProgress ShootCredentialsRotationPhase = "PhaseTwoInProgress"
	// PhaseTwoDone is a constant for the credentials rotation phase describing that Gardener finished all actions required from Gardener side in step two.
	PhaseTwoDone ShootCredentialsRotationPhase = "PhaseTwoDone"
```

Reply:
I understand that it's confusing but personally I don't like "phase one/two" either too much.

What about just renaming `Prepare` -> `Preparing` and `Complete` -> `Completing` so that we end up with `Preparing`, `Prepared`, `Completing` and `Complete`? Would this help preventing the confusion and make it more clear?

/cc @acumino

Reply:
I agree with this proposal.
Semantically it means we have a `prepare` phase that adds trust to the 2nd CA everywhere. In a sense it really prepares the rotation.
Then we have the Completion phase which does the rotation.
I think it is important to have consistent naming. So in the Readme that would be the `Preparation` Phase and the `Completion` phase. We could also think of renaming completing/completed to rotating/rotated. WDYT?

Reply:
> We could also think of renaming completing/completed to rotating/rotated so we have a rotation phase. WDYT?

I think this is again difficult as it implies rotation happens only in the second phase. Actually, it already happens in the first one (client certificates we control would be rotated immediately).